### PR TITLE
main: initialize scheduling group keys before service levels

### PR DESCRIPTION
**Pull request by piodul**
Due to scylladb/seastar#2231, creating a scheduling group and a scheduling group key is not safe to do in parallel. The service level code may attempt to create scheduling groups while the cql_transport::cql_sg_stats scheduling group key is being created.

Until the seastar issue is fixed, move initialization of the cql sg states before service level initialization.

Refs: scylladb/seastar#2231

- [x] No need to backport as it is known to only cause issues in the master branch

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -1021,6 +1021,16 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             dbcfg.commitlog_scheduling_group = make_sched_group("commitlog", "clog", 1000);
             dbcfg.available_memory = memory::stats().total_memory();
 
+            // Make sure to initialize the scheduling group keys at a point where we are sure
+            // that nobody else will be creating service levels (e.g. service levels).
```
Comment (denesb) [inline main.cc:1025]: repetition

**Comment by piodul** [comment]
v2: adjusted the comment to make more sense

File: main.cc (hunk 1 above)
Comment (piodul) [reply main.cc]: Fixed, hopefully the comment makes more sense now

**Review by denesb — APPROVED** [review]

**Comment by piodul** [comment]
image

I didn't add this include, it was already there.

cc: @tchaikov ?

**Comment by tchaikov** [comment]
> image
> 
> I didn't add this include, it was already there.
> 
> cc: @tchaikov ?

should be fixed by #18596. sorry for the inconvenience.

**Review by avikivity — APPROVED** [review]

**Pull request closed without merging**